### PR TITLE
add option to run controller against a namespace

### PR DESCRIPTION
**Pull request by IronPan**
Context: there are some user, especially enterprise customers who wants to use application CRD but don't want to install ClusterRole/ClusterRoleBinding since it has cluster scoped permission. In this case the cluster is a shared multi-user cluster, and cluster admin wants to scope the application controller to scan only the namespace a user owns. 

This change adds a mode to run controller against only a certain namespace.

### CODE REVIEW

**Comment by barney-s** [comment]
Please resolve the conflicts (rebase) and ping again.

File: main.go [hunk 1]
```diff
@@ -44,8 +44,10 @@ func init() {
 }
 
 func main() {
+	var namespace string
 	var metricsAddr string
 	var enableLeaderElection bool
+	flag.StringVar(&namespace, "namespace", "kubeflow", "Namespace within which CRD controller is running.")
```
Comment (barney-s) [inline main.go]: please remove kubeflow. Let the default be ""

File: Dockerfile [hunk 2]
```diff
@@ -45,4 +45,5 @@ FROM gcr.io/distroless/static:latest
 WORKDIR /
 COPY --from=builder /workspace/manager .
 USER nobody
-ENTRYPOINT ["/manager"]
+ENV NAMESPACE ""
+ENTRYPOINT ["/manager", "--namespace=${NAMESPACE}"]
```
Comment (barney-s) [inline Dockerfile]: Please remove the parameter option here. 
Pass it via:
https://github.com/kubernetes-sigs/application/blob/master/config/manager/manager.yaml#L46

File: main.go (hunk 1 above)
Comment (IronPan) [reply main.go]: done

File: Dockerfile (hunk 2 above)
Comment (IronPan) [reply Dockerfile]: done. and i think this file can be configured as needed.

**Pull request merged**